projects/netgrif-components-core/src/lib/navigation/navigation-double-drawer/abstract-navigation-double-drawer.ts

```diff
@@ -373,6 +375,20 @@ export abstract class AbstractNavigationDoubleDrawerComponent implements OnInit,
         return roles;
     }
 
+    protected resolveBannedRoles(filter: Case): Array<RoleAccess> | undefined {
+        const bannedRoles = filter.immediateData.find(f => f.stringId === 'banned_roles')?.options;
+        if (!bannedRoles || Object.keys(bannedRoles).length === 0) return undefined;
+        const roles = [];
+        Object.keys(bannedRoles).forEach(combined => {
```

Comment:
The parsing logic of roles from the configuration is the same as the method `resolveAccessRoles`, to avoid code duplication it should be better to introduce a generic resolving function that is called twice.

Reply:
Thanks for suggestion, you are right, I will solve it.